tests/stubs/userdel-stub/main.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+)
+
+var (
+	flagForce  string
+	flagRemove string
+	flagRoot   string
+)
+
+func main() {
+	flag.StringVar(&flagForce, "force", "", "This option forces the removal of the specified user account")
+	flag.StringVar(&flagRemove, "remove", "", "This option will remove home directory and mail spool")
+	flag.StringVar(&flagRoot, "root", "", "Apply changes in the CHROOT_DIR directory and use the configuration files from the CHROOT_DIR directory")
+
+	flag.Parse()
+
+	if len(flag.Args()) != 1 {
+		fmt.Printf("incorrectly called\n")
+		os.Exit(1)
+	}
+
+	username := flag.Args()[0]
+
+	passwdContents, err := ioutil.ReadFile(path.Join(flagRoot, "/etc/passwd"))
+	if err != nil {
+		fmt.Printf("couldn't open /etc/passwd: %v\n", err)
+		os.Exit(1)
+	}
+	passwdLines := strings.Split(string(passwdContents), "\n")
+	for i, l := range passwdLines {
+		if i == len(passwdLines)-1 {
+			// The last line is empty
+			break
+		}
+		tokens := strings.Split(l, ":")
+		if len(tokens) != 7 {
+			fmt.Printf("scanned incorrect number of items: %d\n", len(tokens))
+			os.Exit(1)
+		}
+
+		if tokens[0] != username {
+			continue
+		}
+
+		passwdLines[i] = l
```

Comment:
Don't we need to append to a new slice here instead of affecting the one we're iterating over?